services/media/service/src/one-time-register-localization-sources.ts

```diff
@@ -0,0 +1,437 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import {
+  createOwnerPgPool,
+  optional,
+  OwnerPgPool,
+} from '@axinom/mosaic-db-common';
+import {
+  RascalConfigBuilder,
+  setupMessagingBroker,
+} from '@axinom/mosaic-message-bus';
+import {
+  LocalizationServiceMultiTenantMessagingSettings,
+  UpsertLocalizationSourceEntityCommand,
+} from '@axinom/mosaic-messages';
+import { Logger, ShutdownAction, sleep } from '@axinom/mosaic-service-common';
+import { singularize } from 'inflection';
+import { all, conditions, parent, select, selectOne } from 'zapatos/db';
+import {
+  collections,
+  collections_images,
+  episodes,
+  episodes_images,
+  movies,
+  movies_images,
+  movie_genres,
+  tvshows,
+  tvshows_images,
+  tvshow_genres,
+} from 'zapatos/schema';
+import { getFullConfig } from './common/config';
+import { requestServiceAccountToken } from './common/utils/token-utils';
+import { LOCALIZATION_COLLECTION_TYPE } from './domains/collections';
+import { buildDisplayTitle } from './domains/common';
+import {
+  LOCALIZATION_MOVIE_GENRE_TYPE,
+  LOCALIZATION_MOVIE_TYPE,
+} from './domains/movies';
+import {
+  LOCALIZATION_EPISODE_TYPE,
+  LOCALIZATION_SEASON_TYPE,
+  LOCALIZATION_TVSHOW_GENRE_TYPE,
+  LOCALIZATION_TVSHOW_TYPE,
+} from './domains/tvshows';
+
+const logger = new Logger({
+  context: 'one-time-register-localization-sources',
+});
+
+/**
+ * This is a one-time migration script to sync localization sources for already
+ * existing movies, movie genres, tvshows, tvshow genres, seasons, episodes, and
+ * collections after the localization functionality is enabled.
+ *
+ * This script must be launched once after updated media service is launched. It
+ * can be launched repeatedly if the need arises, e.g. if new localizable
+ * properties are introduced.
+ *
+ * Entities are processed in batches of 1000, queried by ID, starting with the
+ * oldest. The last batch would also contain all genres.
+ *
+ * Before sending out messages, this script will create corresponding localization
+ * command queues and bindings (if they do not already exist).
+ *
+ * This script only sends out messages and does not modify any database rows in
+ * the media service.
+ */
+async function main(): Promise<void> {
+  logger.log('Starting the migration script...');
+  const config = getFullConfig();
+  const shutdownActions: ShutdownAction[] = [];
+
+  logger.log('Creating messaging and DB components...');
+  const builders = [
+    new RascalConfigBuilder(
+      LocalizationServiceMultiTenantMessagingSettings.UpsertLocalizationSourceEntity,
+      config,
+    ).sendCommand(),
+  ];
+
+  const broker = await setupMessagingBroker({
+    config,
+    builders,
+    logger,
+    shutdownActions,
+  });
+
+  const ownerPool = createOwnerPgPool(
```

Comment:
Does it actually need the owner pool for just reading data?

Reply:
we can use login pool + gql role but then we would need to establish transactions explicitly for each operation (currently the zapatos is doing it). Since this is a migration script - using owner pool is acceptable.